parser.js

```diff
@@ -766,13 +777,60 @@
         group.name = name;
         return group;
       }
+      else if (features.modifiers && str.indexOf("(?") == pos && str[pos+2] != ":") {
+        return parseModifiersGroup();
+      }
       else {
         //      ( Disjunction )
         //      ( ? : Disjunction )
         return parseGroup('(?:', 'ignore', '(', 'normal');
       }
```

Comment:
We should parse `(?sim:...)` and `(?:...)` in the same branch, since they are effectively the same syntax (similarly to how we parse `[]` and `[1, 2, 3]` with the same logic in JS parsers). This will also make sure that we use the same AST for both.

Reply:
```js
    function parseGroup(matchA, typeA, matchB, typeB) {
      var type = null, from = pos;

      if (match(matchA)) {
        type = typeA;
      } else if (match(matchB)) {
        type = typeB;
      } else {
        return false;
      }

      return finishGroup(type, from);
    }
```
But there doesn't seem to be any logic worth sharing. And `parseGroup` is used in multiple places and is not suitable for changing.

The `finishGroup` is currently shared, which ensures that our ASTs are the same.